lib/db.js

```diff
@@ -406,20 +430,9 @@ class LimitDB extends EventEmitter {
       // this partially fixes an issue causing us to collect items in DB indefinitelly
       // those items in general exceed the existing limit of 100 items
       // we return when calling bucket status
-      return bucket.content === bucket.size ? null : bucket;
```

Comment:
The problem was here... if you see the `bucket || {..}` above, there is no `size` parameter in the object literal.

I also made a refactoring to have a simpler codepath that doesn't require a lock with gms, after all is a single operation in leveldb to remove the key, we don't need to do a GET in this case.